src/main/java/org/jenkinsci/plugins/registry/notification/webhook/PushNotification.java

```diff
@@ -134,4 +135,10 @@ public void setCallbackHandler(CallbackHandler callbackHandler) {
     }
 
     abstract public String getRegistryHost();
+
+    public String getDtrEventJSONTypeEventJSONType() { return dtrEventJSONType; }
```

Comment:
This should not be on the base class as it is a specialization in `DockerTrustedRegistryPushNotification`.

Reply:
I put it there instead of casting or instanceOf checks in DockerTrustedRegistryWebHookTest.onFinalized() and EnvContributor.buildEnvironmentFor()  can move it if you think such casting/checking is better.  (It does feel wrong from an OO POV)

Reply:
Well to me it feels more OO to actually utilize inheritance so casting is better in this case imho.

Reply:
Yes, my comment about it "feeling wrong" was meaning the current code feels wrong.  The casting is probably a cleaner solution.   Will update shortly.

Reply:
Moved as per above discussion.